specification/07-theory_operations.adoc

```diff
@@ -360,73 +424,89 @@ The IDE link initial setup must go through the following steps:
-DSM ->> VMM: [DOE] - KEY_EXCHANGE_RSP
-VMM ->> TSM: [COVH] - spdm_resp(KEY_EXCHANGE_RSP)
-TSM ->> TSM: Derive DHE secret
-TSM ->> TSM: Generate SPDM request: FINISH
-TSM ->> VMM: [COVH] - spdm_req(FINISH)
-VMM ->> DSM: [DOE] - SPDM_FINISH
-DSM ->> VMM: [DOE] - FINISH_RSP
-VMM ->> TSM: [COVH] - spdm_resp(FINISH_RSP)
-
-Note over TSM,DSM: TSM: SPDM Session Created
-
-%% TODO Add the IDE sequence...
-Note over TSM,DSM: IDE Link setup starts here...
+VMM ->> VMM: Find an available IDE stream ID
+VMM ->> Device: Program IDE extended capability
+
+VMM ->> TSM: [COVH] - sbi_covh_connect_device(device_id, stream_id)
+
+Note over TSM,DSM: TSM: SPDM Connection Initialization
+
+Note over TSM,DSM: TSM: SPDM Connection Created
+
+TSM ->> TSM: Generate 6 IDE keys [(Rx, Tx) * sub-streams(P, NP, C)]
+
+Loop 6 times - TSM programs IDE keys into Device (For the given stream ID)
+     TSM ->> TSM: Generate SPDM request IDE_KM_KEY_PROG
+     TSM ->> VMM: [COVEH] - spdm_req(IDE_KM_KEY_PROG)
+     VMM ->> DSM: [DOE] - SPDM_IDE_KM_KEY_PROG
+     DSM ->> Device: Program  IDE Key
+     DSM ->> VMM: [DOE] - SPDM_IDE_KM_KP_ACK
+     VMM ->> TSM: [COVEH] - spdm_resp(IDE_KM_KP_ACK)
+end
+
+Loop 6 times - TSM programs IDE keys into Root Port through RoT (For the given stream ID)
+     TSM ->> TSM: Generate SPDM request IDE_KM_KEY_PROG
+     TSM ->> VMM: [COVEH] - spdm_req(IDE_KM_KEY_PROG)
+     VMM ->> RoT: [DOE] - SPDM_IDE_KM_KEY_PROG
+     RoT ->> RootPort: Program  IDE Key
+     RoT ->> VMM: [DOE] - SPDM_IDE_KM_KP_ACK
+     VMM ->> TSM: [COVEH] - spdm_resp(IDE_KM_KP_ACK)
+end
+
+Loop 3 times (For each sub-stream)
+     TSM ->> TSM: Generate SPDM request IDE_KM_K_SET_GO
+     TSM ->> VMM: [COVEH] - spdm_req(IDE_KM_K_SET_GO)
+     VMM ->> DSM: [DOE] - SPDM_IDE_KM_K_SET_GO
+     DSM ->> Device: Trigger Tx IDE
+     DSM ->> VMM: [DOE] - SPDM_IDE_KM_K_GOSTOP_ACK
+     VMM ->> TSM: [COVEH] - spdm_resp(IDE_KM_K_GOSTOP_ACK)
+end
+
+Loop 3 times (For each sub-stream)
+     TSM ->> TSM: Generate SPDM request IDE_KM_K_SET_GO
+     TSM ->> VMM: [COVEH] - spdm_req(IDE_KM_K_SET_GO)
+     VMM ->> RoT: [DOE] - SPDM_IDE_KM_K_SET_GO
+     RoT ->> Device: Trigger Tx IDE
+     RoT ->> VMM: [DOE] - SPDM_IDE_KM_K_GOSTOP_ACK
+     VMM ->> TSM: [COVEH] - spdm_resp(IDE_KM_K_GOSTOP_ACK)
+end
+
+Loop 3 times (For each sub-stream)
+     TSM ->> TSM: Generate SPDM request IDE_KM_K_SET_GO
+     TSM ->> VMM: [COVEH] - spdm_req(IDE_KM_K_SET_GO)
+     VMM ->> DSM: [DOE] - SPDM_IDE_KM_K_SET_GO
+     DSM ->> Device: Trigger Rx IDE
+     DSM ->> VMM: [DOE] - SPDM_IDE_KM_K_GOSTOP_ACK
+     VMM ->> TSM: [COVEH] - spdm_resp(IDE_KM_K_GOSTOP_ACK)
+end
+
+Loop 3 times (For each sub-stream)
+     TSM ->> TSM: Generate SPDM request IDE_KM_K_SET_GO
+     TSM ->> VMM: [COVEH] - spdm_req(IDE_KM_K_SET_GO)
+     VMM ->> RoT: [DOE] - SPDM_IDE_KM_K_SET_GO
+     RoT ->> Device: Trigger Rx IDE
+     RoT ->> VMM: [DOE] - SPDM_IDE_KM_K_GOSTOP_ACK
+     VMM ->> TSM: [COVEH] - spdm_resp(IDE_KM_K_GOSTOP_ACK)
+end
 
```

Comment:
we should add 2 steps here to enable IDE.
1) Enable IDE on device side.
2) Enable IDE on Root Port side.

Reply:
Done.

Reply:
The order should be reversed.

First to enable Device, then enable Root Port.

Reply:
Fixed as well, thanks.